Author: Alex Yam <alex@alexyam.com>

MutableSet and Iterable moved from collections to collections.abc in Python 3.10

--- a/pylib/gyp/common.py
+++ b/pylib/gyp/common.py
@@ -494,7 +494,7 @@
 
 
 # Based on http://code.activestate.com/recipes/576694/.
-class OrderedSet(collections.MutableSet):
+class OrderedSet(collections.abc.MutableSet):
   def __init__(self, iterable=None):
     self.end = end = []
     end += [None, end, end]         # sentinel node for doubly linked list
     
--- a/test/lib/TestGyp.py
+++ b/test/lib/TestGyp.py
@@ -496,7 +496,7 @@
     if status is None:
       kw['status'] = None
     else:
-      if not isinstance(status, collections.Iterable): status = (status,)
+      if not isinstance(status, collections.abc.Iterable): status = (status,)
       kw['status'] = list(itertools.chain((0,), status))
     self.cmake_build(gyp_file, target, **kw)
     kw['status'] = status
